weaver/execute.py

```diff
@@ -1,20 +1,75 @@
+import logging
 from typing import TYPE_CHECKING
 
+from pyramid.httpexceptions import HTTPBadRequest
+
 from weaver.base import Constants
+from weaver.utils import get_header, parse_kvp
 
 if TYPE_CHECKING:
-    from typing import List
+    from typing import List, Optional, Tuple, Union
+
+    from weaver.datatype import Job
+    from weaver.typedefs import AnyHeadersContainer, HeadersType, Literal
+
+    ExecutionModeAutoType = Literal["auto"]
+    ExecutionModeAsyncType = Literal["async"]
+    ExecutionModeSyncType = Literal["sync"]
+    AnyExecuteMode = Union[
+        ExecutionModeAutoType,
+        ExecutionModeAsyncType,
+        ExecutionModeSyncType,
+    ]
+    ExecuteControlOptionAsyncType = Literal["async-execute"]
+    ExecuteControlOptionSyncType = Literal["sync-execute"]
+    AnyExecuteControlOption = Union[
+        ExecuteControlOptionAsyncType,
+        ExecuteControlOptionSyncType,
+    ]
+    ExecuteReturnPreferenceMinimalType = Literal["minimal"]
+    ExecuteReturnPreferenceRepresentationType = Literal["representation"]
+    AnyExecuteReturnPreference = Union[
+        ExecuteReturnPreferenceMinimalType,
+        ExecuteReturnPreferenceRepresentationType,
+    ]
+    ExecuteResponseDocumentType = Literal["document"]
+    ExecuteResponseRawType = Literal["raw"]
+    AnyExecuteResponse = Union[
+        ExecuteResponseDocumentType,
+        ExecuteResponseRawType,
+    ]
+    ExecuteTransmissionModeReferenceType = Literal["reference"]
+    ExecuteTransmissionModeValueType = Literal["value"]
+    AnyExecuteTransmissionMode = Union[
+        ExecuteTransmissionModeReferenceType,
+        ExecuteTransmissionModeValueType,
+    ]
+    # pylint: disable=C0103,invalid-name
```

Comment:
This is explicit to remove warnings from `pylint` that wants something like `ExecuteCollectionFormatTypeOgcCoverage` rather than `ExecuteCollectionFormatType_OGC_COVERAGE`. Since there's mixed use of acronyms and names in there, having the `_<name>` suffix is more readable than typical naming rules.